server/routers/api-router.js

```diff
@@ -253,7 +249,7 @@ router.get("/api/badge/:id/uptime/:duration?", cache("5 minutes"), async (reques
             badgeValues.color = badgeConstants.naColor;
         } else {
             const uptimeCalculator = await UptimeCalculator.getUptimeCalculator(requestedMonitorId);
-            const uptime = overrideValue ?? uptimeCalculator.getDataByDuration(requestedDuration).uptime;
+            const uptime = overrideValue ?? uptimeCalculator.getDataByDuration(`${requestedDuration}h`).uptime;
```

Comment:
please move this back up.
Currently, I don't think you are handling this correctly.

Think about cases where I supply `30d` => this is now handled as `30dh`..

Reply:
We have not yet looked deeply at that part of the UI.
Related #4500